src/wallet/wallet_rpc_server.cpp

```diff
@@ -73,6 +73,17 @@ using namespace epee;
     } \
   } while(0)
 
+#define CHECK_BACKGROUND_SYNC_ENABLED() \
+  do \
+  { \
+    if (m_wallet->background_sync_enabled()) \
+    { \
+      er.code = WALLET_RPC_ERROR_CODE_BACKGROUND_SYNC_ENABLED; \
+      er.message = "This command is disabled while background sync is enabled. Disable background sync to use this command."; \
+      return false; \
+    } \
+  } while(0)
+
```

Comment:
There are some additional methods that could use a call to this, in addition to the 6 that already have it, to note `on_rescan_spent`, `on_rescan_blockchain` and `on_sign` and maybe all the transfer and sweep methods, but it's probably not worth it to add such checks because `wallet2` will stop you anyway reliably if the spend secret key is needed but not available.

Reply:
Agree with all except I think `on_rescan_blockchain` should be ok to call (see how the tests use it)